server/src/test/java/org/opensearch/indices/ShardLimitValidatorTests.java

```diff
@@ -214,6 +214,19 @@ public void testNonSystemIndexCreationFailsWithMaxShardLimitOnCluster() {
         );
     }
 
+    public void testMaxValueOfMaxShardLimitOnCluster() {
+        final int maxShardLimitOnCluster = 2_000_000;
+        Settings limitOnlySettings = Settings.builder().put(SETTING_CLUSTER_MAX_SHARDS_PER_NODE.getKey(), maxShardLimitOnCluster).build();
+        final IllegalArgumentException exception = expectThrows(
+            IllegalArgumentException.class,
+            () -> createTestShardLimitService(limitOnlySettings)
+        );
+        assertEquals(
```

Comment:
Does this fail with something like the `Validation Failed: 1: this action would add [2] total shards, but this cluster currently has [30101]/[-1089934592] maximum shards open;` error without this fix?

Reply:
Yes, if the user set the cluster.max_shards_per_node too large(such as 500_000_000), it will hit this case. 

In our product, user will manage the cluster directly, he is not familiar with the parameter. To save trouble, 500_000_000 is set, which caused the rollover index to be unable to be allocated the next day.